### PR TITLE
feat(default-theme): display 400 errors right from the API

### LINKED ISSUE #555: Make the registration process consistent in terms of API errors notofications
Acceptance criteria:
* [ ] -  the API response is parsed by 
```
{ getMessagesFromErrorsArray} from '@shopware-pwa/helpers' 
```
and output is displayed above the form

* [ ] - the list of registration forms is listed here
* [ ] - it's taken under consideration to use registration/form factory if it's easy to implement

### PR DESCRIPTION
**Pull request by mkucmus**
closes #555 

- fixes my-account update forms issues
- displaying appropriate error messages that come from API in case of 400 (bad request)

### Checklist

- [x] I followed [contributing](https://github.com/DivanteLtd/shopware-pwa/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: packages/default-theme/components/forms/SwEmail.vue [hunk 1]
```diff
@@ -67,44 +67,55 @@ import { validationMixin } from 'vuelidate'
...
   setup() {
-    const { user, error, updateEmail, refreshUser } = useUser()
+    const { user, error: userError, updateEmail, refreshUser } = useUser()
     return {
       refreshUser,
       updateEmail,
       user,
-      error
+      userError,
+      getMessagesFromErrorsArray
```
Comment (patzick) [inline packages/default-theme/components/forms/SwEmail.vue:84]:
```suggestion
      userErrorMessages
```

File: packages/default-theme/components/forms/SwEmail.vue [hunk 2]
```diff
@@ -67,44 +67,55 @@ import { validationMixin } from 'vuelidate'
...
 
 export default {
-  name: 'MyProfile',
+  name: 'EmailChange',
   components: { SfInput, SfButton, SfAlert },
   mixins: [validationMixin],
   props: {},
   setup() {
-    const { user, error, updateEmail, refreshUser } = useUser()
+    const { user, error: userError, updateEmail, refreshUser } = useUser()
```
Comment (patzick) [inline packages/default-theme/components/forms/SwEmail.vue:78]: we can do computation here, getMessagesFromErrorsArray checks for nulls so we can pass just argument :)

```suggestion
    const { user, error: userError, updateEmail, refreshUser } = useUser()
    const userErrorMessages = computed(() => getMessagesFromErrorsArray(userError.value.message))
```

File: packages/default-theme/components/forms/SwEmail.vue [hunk 3]
```diff
@@ -67,44 +67,55 @@ import { validationMixin } from 'vuelidate'
...
   },
+  computed: {
+    useUserErrorMessages() {
+      // all the 400 errors are in a raw format stright from the API - to be extracted easily depeding on needs.
+      return (
+        this.userError &&
+        this.getMessagesFromErrorsArray(this.userError.message)
+      )
+    }
+  },
```
Comment (patzick) [inline packages/default-theme/components/forms/SwEmail.vue:102]: then we don't need this computed section in components

File: packages/default-theme/components/forms/SwEmail.vue [hunk 4]
```diff
@@ -67,44 +67,55 @@ import { validationMixin } from 'vuelidate'
...
   methods: {
     async invokeUpdate() {
       const emailChanged = await this.updateEmail({
         email: this.email,
-        confirmEmail: this.confirmEmail,
+        emailConfirmation: this.confirmEmail,
```
Comment (mkucmus) [inline packages/default-theme/components/forms/SwEmail.vue:107]: @Michal-Dziedzinski tu jest fix

**Comment by Michal-Dziedzinski** [comment]
also closes; #416

**Comment by mkucmus** [comment]
> On iOS during registration there is a problem with scrolling to the place where 'Don't have and account yet? Register today? ' is written.
> RejestracjaiOS
> Tested on browserstack.

thanks @akrajna :+1:, could you create a new issue regarding this?

**Comment by akrajna** [comment]
@mkucmus the issue have been created #591 :-)

**Comment by mkucmus** [comment]
> @mkucmus the issue have been created #591 :-)

@patzick as you can see - it's ready :+1:

**Pull request merged**